recommender/recommender.py

```diff
@@ -494,55 +524,59 @@ def edit_resource(self, data, _suffix=''):
                 result[old_resource_content_field]: the content of the resource before edited
                 result[resource_content_field]: the content of the resource after edited
         """
-        resource_id = data['id']
-        idx = self.get_entry_index(resource_id, self.recommendations)
-        if idx not in range(0, len(self.recommendations)):
+        resource_id = stem_url(data['id'])
+        if resource_id not in self.recommendations:
             msg = 'The selected resource is not existing'
             return self.error_handler(msg, 'edit_resource', resource_id)
 
         result = {}
         result['id'] = resource_id
+        result['old_id'] = resource_id
+
         for field in self.resource_content_fields:
-            result['old_' + field] = self.recommendations[idx][field]
+            result['old_' + field] = self.recommendations[resource_id][field]
             if data[field] == "":
-                result[field] = self.recommendations[idx][field]
+                result[field] = self.recommendations[resource_id][field]
             else:
                 result[field] = data[field]
 
-        if not(self.check_redundancy(self.recommendations[idx]['url'],
-                                     data['url'])):
+        edited_resource_id = stem_url(data['url'])
+        if edited_resource_id != resource_id:
             # check url for redundancy
-            for recommendation in self.recommendations:
-                if self.check_redundancy(recommendation['url'], data['url']):
-                    result['error'] = ('The resource you are attempting to ' +
-                                       'provide has already existed')
-                    for field in self.resource_content_fields:
-                        result['dup_' + field] = recommendation[field]
-                    result['dup_id'] = recommendation['id']
-                    result['Success'] = False
-                    tracker.emit('edit_resource', result)
-                    return result
-                
+            if edited_resource_id in self.recommendations:
+                result['error'] = ('The resource you are attempting to ' +
+                                   'provide has already existed')
```

Comment:
Seems like you're duplicating some logic you had earlier. Can you refactor to avoid duplicating code?